python/pyarrow/util.py

```diff
@@ -228,3 +228,40 @@ def _break_traceback_cycle_from_frame(frame):
         # us visit the outer frame).
         refs = gc.get_referrers(frame)
     refs = frame = this_frame = None
+
+
+def download_tzdata_on_windows():
+    """
+    Download and extract 2021e IANA timezone database into
+    the expected location in the Downloads directory.
+    """
+    if sys.platform != 'win32':
+        raise TypeError(f"Timezone database is already provided by {sys.platform}")
+
+    import requests
```

Comment:
Given we don't require requests at the moment as a dependency, we could also use the stdlib `urllib.request.urlopen` (it's no problem for a util function like this to have additional dependency requirements, but if there is an easy stdlib alternative, let's just use that to avoid this issue)